dragonfly/item/flint_and_steel.go

```diff
@@ -0,0 +1,43 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/df-mc/dragonfly/dragonfly/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// FlintAndSteel is an item used to light blocks on fire.
+type FlintAndSteel struct{}
+
+// MaxCount ...
+func (f FlintAndSteel) MaxCount() int {
+	return 1
+}
+
+// DurabilityInfo ...
+func (f FlintAndSteel) DurabilityInfo() DurabilityInfo {
+	return DurabilityInfo{
+		MaxDurability: 65,
+		BrokenItem:    simpleItem(Stack{}),
+	}
+}
+
+// UseOnBlock ...
+func (f FlintAndSteel) UseOnBlock(pos world.BlockPos, face world.Face, _ mgl64.Vec3, w *world.World, _ User, ctx *UseContext) bool {
+	ctx.DamageItem(1)
+	w.PlaySound(pos.Vec3(), sound.Ignite{})
```

Comment:
I don't think a flint and steel is supposed to lose durability and play a sound when it doesn't actually do anything.